### DISCUSSION TITLE
Haskell bindings

### DISCUSSION THREAD

**Discussion by velveteer** [simdjson/simdjson#1748] [Show and tell]
I've been working on some Haskell bindings to the On Demand API, the repo is here: https://github.com/velveteer/hermes/

There's no official release yet, but I've tested it in production code with pretty good results. The performance benefit for partial decoding is impressive, and I am still trying to squeeze more performance out of it, although the Haskell FFI has its limits. 

I am not a seasoned C++ programmer, so any feedback is very appreciated. With any luck I will eventually release it on Hackage (Haskell package registry) and let y'all know about it. Thanks for writing such a great library and the terrific documentation, I couldn't have made much progress at all without the information generously provided by the simdjson authors and contributors.

**Comment by lemire** [comment]
@jkeiser : It looks like @velveteer used On Demand, so it could be of interest to you.

**Comment by velveteer** [comment]
`hermes-json` is now officially released to the Hackage package index: https://hackage.haskell.org/package/hermes-json

It's not a full binding to the simdjson API, but there's enough in this release for Haskell users to start taking advantage of simdjson's performance, especially for partial decoding. 

Thanks again for the amazing work on this library and the documentation.

**Reply by lemire** [reply]
Fantastic.
